packages/members-api/index.js

```diff
@@ -239,6 +240,46 @@ module.exports = function MembersApi({
         }
     });
 
+    middleware.cancelSubscription.use(ensureStripe, body.json(), async function (req, res) {
+        const identity = req.body.identity;
+        const subscriptionId = req.params.id;
```

Comment:
Is there a reason this isn't on the body?

Reply:
Yes :)  It's present in request parameters define like [this](https://github.com/TryGhost/Ghost/pull/11434/files#diff-fe969debf1e1dcf0c7a61daf62011921R24
): 
```
apiApp.delete('/subscriptions/:id', (req, res, next) => membersService.api.middleware.cancelSubscription(req, res, next));
```
My thinking was: we are not using the "frame" here, which merges all the data inputs query/params/options, so why complicate? :man_shrugging: 

Theoretically, we could do something as we do in the frame:
```javascript
if (Object.prototype.hasOwnProperty.call(this.original, 'params')) {
    Object.assign(this.options, _.pick(this.original.params, apiConfig.options));
}
```

But don't think it's worth cherry-picking parts of the behavior from the core and we want to be consistent should rather implement the frame on top of these endpoints.

What's your opinion?

Reply:
I just saw the companion PR and saw the `/:id` stuff - yeah makes total sense - it's a but of a shame that this middleware dictates what the URL should look like - e.g. 

You can't use this middleware with a route like `/subscription/:subscription_id` - which kinda sucks - BUT - can't see a better way without having to change our url and break consistency in the API